06_HelloGraphicsQueue/main.cpp

```diff
@@ -2,80 +2,436 @@
+
+		IImage::SBufferCopy bufferCopy;
+		bufferCopy.bufferImageHeight = smallImgExtent.height;
+		bufferCopy.bufferRowLength = smallImgExtent.width;
+		bufferCopy.bufferOffset = 0u;
+		bufferCopy.imageExtent = { smallImgExtent.width, smallImgExtent.height, 1u };
+		bufferCopy.imageSubresource = subresourceLayers;
+
+		SImageBlit firstImgBlit{};
+		{
+			asset::VkOffset3D dstOffset0;
+			dstOffset0.x = bigImgExtent.width / 2u - smallImgExtent.width / 2u;
+			dstOffset0.y = bigImgExtent.height / 2u - smallImgExtent.height / 2u;
+			dstOffset0.z = 0u;
+
+			asset::VkOffset3D dstOffset1;
+			dstOffset1.x = dstOffset0.x + smallImgExtent.width;
+			dstOffset1.y = dstOffset0.y + smallImgExtent.height;
+			dstOffset1.z = 1u;
+
+			firstImgBlit.srcSubresource = subresourceLayers;
+			firstImgBlit.srcOffsets[0] = { 0u, 0u, 0u };
+			firstImgBlit.srcOffsets[1] = { smallImgExtent.width, smallImgExtent.height, 1u };
+			firstImgBlit.dstSubresource = subresourceLayers;
+			firstImgBlit.dstOffsets[0] = dstOffset0;
+			firstImgBlit.dstOffsets[1] = dstOffset1;
+		}
+
+		SImageBlit secondImgBlit{};
+		{
+			secondImgBlit.srcSubresource = subresourceLayers;
+			secondImgBlit.srcOffsets[0] = { 0u, 0u, 0u };
+			secondImgBlit.srcOffsets[1] = { bigImgExtent.width, bigImgExtent.height, 1u };
+			secondImgBlit.dstSubresource = subresourceLayers;
+			secondImgBlit.dstOffsets[0] = { 0u, 0u, 0u };
+			secondImgBlit.dstOffsets[1] = { smallImgExtent.width, smallImgExtent.height, 1u };
+		}
 
-			core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
-			if (!m_device->createCommandBuffers(nullptr,IGPUCommandBuffer::EL_PRIMARY,1,&cmdbuf))
+		core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
+		{
+			core::bitflag<IGPUCommandPool::E_CREATE_FLAGS> flags = static_cast<IGPUCommandPool::E_CREATE_FLAGS>(IGPUCommandPool::E_CREATE_FLAGS::ECF_TRANSIENT_BIT);
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(queue->getFamilyIndex(), flags);
+			if (!m_device->createCommandBuffers(cmdpool.get(), IGPUCommandBuffer::EL_PRIMARY, 1, &cmdbuf))
 				return false;
+		}
+
+		IGPUQueue::SSubmitInfo submitInfo{};
+		submitInfo.commandBufferCount = 1;
+		submitInfo.commandBuffers = &cmdbuf.get();
+
+		IGPUCommandBuffer::SImageMemoryBarrier imgLayoutTransitionBarrier = constructDefaultInitializedImageBarrier();
+		imgLayoutTransitionBarrier.newLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
+		imgLayoutTransitionBarrier.barrier.dstAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
+
+		IGPUCommandBuffer::SImageMemoryBarrier imgClearBarrier = constructDefaultInitializedImageBarrier();
+		imgClearBarrier.oldLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
+		imgClearBarrier.barrier.srcAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
+
+		constexpr SClearColorValue red = { .float32{1.0f, 0.0f, 0.0f, 1.0f} };
+		constexpr SClearColorValue blue = { .float32{0.0f, 0.0f, 1.0f, 1.0f} };
```

Comment:
explain why we're using this union member isntead of others (colorformat stuff I explained to you)